examples/static_graph_optimizations/cifar/train_cifar_custom_loop.py

```diff
@@ -101,50 +97,61 @@ def main():
     optimizer.add_hook(chainer.optimizer.WeightDecay(5e-4))
 
     train_iter = chainer.iterators.SerialIterator(train, args.batchsize)
-    test_iter = chainer.iterators.SerialIterator(test, args.batchsize,
-                                                 repeat=False, shuffle=False)
-
-    sum_accuracy = 0
-    sum_loss = 0
-
-    while train_iter.epoch < args.epoch:
-        batch = train_iter.next()
-        # Reduce learning rate by 0.5 every 25 epochs.
-        if train_iter.epoch % 25 == 0 and train_iter.is_new_epoch:
-            optimizer.lr *= 0.5
-            print('Reducing learning rate to: {}'.format(optimizer.lr))
-
-        x_array, t_array = convert.concat_examples(batch, device)
-        x = chainer.Variable(x_array)
-        t = chainer.Variable(t_array, requires_grad=False)
-        optimizer.update(model, x, t)
-        sum_loss += float(model.loss.array) * len(t)
-        sum_accuracy += float(model.accuracy.array) * len(t)
-
-        if train_iter.is_new_epoch:
-            print('epoch: {}'.format(train_iter.epoch))
-            print('train mean loss: {}, accuracy: {}'.format(
-                sum_loss / train_count, sum_accuracy / train_count))
-            # evaluation
-            sum_accuracy = 0
-            sum_loss = 0
-            model.predictor.train = False
-            # It is good practice to turn off train mode during evaluation.
-            with configuration.using_config('train', False):
-                for batch in test_iter:
-                    x_array, t_array = convert.concat_examples(batch, device)
-                    x = chainer.Variable(x_array)
-                    t = chainer.Variable(t_array, requires_grad=False)
-                    loss = model(x, t)
-                    sum_loss += float(loss.array) * len(t)
-                    sum_accuracy += float(model.accuracy.array) * len(t)
-
-            test_iter.reset()
-            model.predictor.train = True
-            print('test mean  loss: {}, accuracy: {}'.format(
-                sum_loss / test_count, sum_accuracy / test_count))
-            sum_accuracy = 0
-            sum_loss = 0
+    test_iter = chainer.iterators.SerialIterator(
+        test, args.batchsize, repeat=False, shuffle=False)
+
+    def run():
```

Comment:
How about making it a module-level function, instead of nested one?
Module-level functions are more easily understood for Python beginners I think.

The name `run` looks too simple.
How about `run_train_loop` or something like that?

Reply:
Changed as suggested.